jetstream/allocation.py

```diff
@@ -255,32 +257,48 @@ def get_user_allocations(self, username, include_expired=False, raise_exception=
 
 
 
-def get_or_create_allocation_source(api_allocation, update_source=False):
+def get_or_create_allocation_source(api_allocation):
     try:
         source_name = "%s" % (api_allocation['project'],)
         source_id = api_allocation['id']
         compute_allowed = int(api_allocation['computeAllocated'])
     except (TypeError, KeyError, ValueError):
         raise TASAPIException("Malformed API Allocation - Missing keys in dict: %s" % api_allocation)
 
+    payload = {
+        'allocation_source_name': source_name,
+        'compute_allowed': compute_allowed,
+        'start_date':api_allocation['start'],
+        'end_date':api_allocation['end']
+    }
+
+    # Hash name and source_id to check for renewed Allocation Source
     try:
-        source = AllocationSource.objects.get(
-            source_id=source_id
-        )
-        if update_source:
-            if compute_allowed != source.compute_allowed:
-                #FIXME: Here would be a *great* place to create a new event to "ignore" all previous allocation_source_`threshold_met/threshold_enforced`
-                source.compute_allowed = compute_allowed
-            source.name = source_name
-            source.save()
-        return source, False
-    except AllocationSource.DoesNotExist:
-        source = AllocationSource.objects.create(
-            name=source_name,
-            compute_allowed=compute_allowed,
-            source_id=source_id
-        )
-        return source, True
+        created_event_key = 'sn=%s,si=%s,ev=%s,dc=jetstream,dc=atmosphere' % (
+            source_name, source_id, 'allocation_source_created_or_renewed')
```

Comment:
I'm 👍 on a method defined. 

I would consider having that method use a format string definition of this as well